crates/router/src/connector/datatrans.rs

```diff
@@ -357,10 +415,13 @@ impl ConnectorIntegration<api::Capture, types::PaymentsCaptureData, types::Payme
         event_builder: Option<&mut ConnectorEvent>,
         res: Response,
     ) -> CustomResult<types::PaymentsCaptureRouterData, errors::ConnectorError> {
-        let response: datatrans::DatatransPaymentsResponse = res
-            .response
-            .parse_struct("Datatrans PaymentsCaptureResponse")
-            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        let response = if res.response.is_empty() {
+            datatrans::DataTransCaptureResponse::Empty
```

Comment:
Do we have a possibility to get an empty response for flows other than capture and cancel, also for response with http status code other than 2xx.

Reply:
We only get an empty response for capture and cancel requests. For the rest of the requests, we receive a response body along with a 2xx code for successful operations.